src/Microsoft.Health.Fhir.Core/Features/Search/ContinuationToken.cs

```diff
@@ -0,0 +1,104 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search
+{
+    /// <summary>
+    /// Represents the search continuation token.
+    /// </summary>
+    public class ContinuationToken
+    {
+        // the token is an array.
+        private object[] _tokens;
+
+        public ContinuationToken(object[] tokens)
+        {
+            _tokens = tokens;
+        }
+
+        public long ResourceSurrogateId
+        {
+            get
+            {
+                return (long)_tokens[^1];
+            }
+
+            set
+            {
+                _tokens[^1] = value;
+            }
+        }
+
+        // Currently only a single sort is implemented
+        public string SortValue
+        {
+            get
+            {
+                return _tokens.Length > 1 ? _tokens[0] as string : null;
+            }
+        }
+
+        public string ToJson()
+        {
+            return JArray.FromObject(_tokens).ToString();
+        }
+
+        public override string ToString()
+        {
+            return ToJson();
+        }
+
+        public static ContinuationToken FromString(string json)
```

Comment:
This approach is not particularly efficient. There are two improvements that could be made.

|         Method |     Mean |     Error |    StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|--------------- |---------:|----------:|----------:|-------:|------:|------:|----------:|
|        Current | 4.168 us | 0.0775 us | 0.0762 us | 0.6180 |     - |     - |    3264 B |
|     Simplified | 1.811 us | 0.0297 us | 0.0445 us | 0.5398 |     - |     - |    2824 B |
| SystemTextJson | 1.140 us | 0.0219 us | 0.0269 us | 0.0458 |     - |     - |     240 B |


The "simplified" option looks like this:
``` csharp
private static JsonSerializerSettings settings = new JsonSerializerSettings
{
    DateParseHandling = DateParseHandling.None,
};


[Benchmark]
public object[] Simplified()
{
    if (long.TryParse(json, NumberStyles.None, CultureInfo.InvariantCulture, out var sid))
    {
        return new object[] { sid };
    }

    return JsonConvert.DeserializeObject<object[]>(json, settings);
}
```

Faster still, and with a lot fewer allocations, is an implementation based on System.Text.Json:
``` csharp
private static readonly JsonSerializerOptions options = new JsonSerializerOptions() { Converters = { new MyConverter() } };

[Benchmark]
public object[] SystemTextJson()
{
    if (long.TryParse(json, NumberStyles.None, CultureInfo.InvariantCulture, out var sid))
    {
        var result = new object[] { sid };
        return result;
    }

    return System.Text.Json.JsonSerializer.Deserialize<object[]>(json, options);
}

public class MyConverter : System.Text.Json.Serialization.JsonConverter<object>
{
    public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) => reader.TokenType switch
    {
        JsonTokenType.String => reader.GetString(),
        JsonTokenType.Number => reader.GetInt64(),
        _ => throw new NotSupportedException()
    };

    public override void Write(Utf8JsonWriter writer, object value, JsonSerializerOptions options) => throw new NotImplementedException();
}
```

Reply:
@johnstairs ok, I used the SystemTextJson as you suggested